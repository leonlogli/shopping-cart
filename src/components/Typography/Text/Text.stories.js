import React from 'react'

import Text from './index'

export default {
  title: 'Text',
}

export const text = () => (
  <div style={{ padding: '20px 100px' }}>
    <Text variant="subtitle1" gutterBottom>
      subtitle1. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos
      blanditiis tenetur
    </Text>
    <Text variant="subtitle2" gutterBottom>
      subtitle2. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos
      blanditiis tenetur
    </Text>
    <Text variant="body1" gutterBottom>
      body1. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos
      blanditiis tenetur unde suscipit, quam beatae rerum inventore consectetur,
      neque doloribus, cupiditate numquam dignissimos laborum fugiat deleniti?
      Eum quasi quidem quibusdam.
    </Text>
    <Text variant="body2" gutterBottom>
      body2. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quos
      blanditiis tenetur unde suscipit, quam beatae rerum inventore consectetur,
      neque doloribus, cupiditate numquam dignissimos laborum fugiat deleniti?
      Eum quasi quidem quibusdam.
    </Text>
    <Text variant="button" display="block" gutterBottom>
      button text
    </Text>
    <Text variant="caption" display="block" gutterBottom>
      caption text
    </Text>
    <Text variant="overline" display="block" gutterBottom>
      overline text
    </Text>
  </div>
)
