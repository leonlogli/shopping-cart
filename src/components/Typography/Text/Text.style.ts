import styled from 'styled-components'
import Typography from '@material-ui/core/Typography'

export const StyledText = styled(Typography)``

export default StyledText
