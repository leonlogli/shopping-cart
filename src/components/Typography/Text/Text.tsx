import React from 'react'
import { TypographyProps } from '@material-ui/core/Typography'

import StyledText from './Text.style'

export type TextProps = Omit<TypographyProps, 'variant'> & {
  /**
   * Applies the theme typography styles.
   */
  variant?:
    | 'subtitle1'
    | 'subtitle2'
    | 'body1'
    | 'body2'
    | 'caption'
    | 'button'
    | 'overline'
    | 'srOnly'
    | 'inherit'
  /**
   * The component used for the root node. Either a string to use a HTML element or a component.
   */
  component?: keyof JSX.IntrinsicElements | React.ComponentType<any>
}

const Text = ({ children, ...other }: TextProps) => (
  <StyledText {...other}>{children}</StyledText>
)

export { Text }
export default Text
