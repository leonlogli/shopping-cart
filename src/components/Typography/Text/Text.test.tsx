import React from 'react'
import { mount } from 'enzyme'
import Text from './index'

describe('<Text />', () => {
  const wrapper = mount(<Text>text</Text>)

  it('should have the correct children', () => {
    expect(wrapper.props().children).toBe('text')
  })

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
