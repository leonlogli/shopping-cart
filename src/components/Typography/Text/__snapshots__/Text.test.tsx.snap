// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should match a snapshot 1`] = `
<Text>
  <Styled(WithStyles(ForwardRef(Typography)))>
    <WithStyles(ForwardRef(Typography))
      className=""
    >
      <ForwardRef(Typography)
        className=""
        classes={
          Object {
            "alignCenter": "MuiTypography-alignCenter",
            "alignJustify": "MuiTypography-alignJustify",
            "alignLeft": "MuiTypography-alignLeft",
            "alignRight": "MuiTypography-alignRight",
            "body1": "MuiTypography-body1",
            "body2": "MuiTypography-body2",
            "button": "MuiTypography-button",
            "caption": "MuiTypography-caption",
            "colorError": "MuiTypography-colorError",
            "colorInherit": "MuiTypography-colorInherit",
            "colorPrimary": "MuiTypography-colorPrimary",
            "colorSecondary": "MuiTypography-colorSecondary",
            "colorTextPrimary": "MuiTypography-colorTextPrimary",
            "colorTextSecondary": "MuiTypography-colorTextSecondary",
            "displayBlock": "MuiTypography-displayBlock",
            "displayInline": "MuiTypography-displayInline",
            "gutterBottom": "MuiTypography-gutterBottom",
            "h1": "MuiTypography-h1",
            "h2": "MuiTypography-h2",
            "h3": "MuiTypography-h3",
            "h4": "MuiTypography-h4",
            "h5": "MuiTypography-h5",
            "h6": "MuiTypography-h6",
            "noWrap": "MuiTypography-noWrap",
            "overline": "MuiTypography-overline",
            "paragraph": "MuiTypography-paragraph",
            "root": "MuiTypography-root",
            "srOnly": "MuiTypography-srOnly",
            "subtitle1": "MuiTypography-subtitle1",
            "subtitle2": "MuiTypography-subtitle2",
          }
        }
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          text
        </p>
      </ForwardRef(Typography)>
    </WithStyles(ForwardRef(Typography))>
  </Styled(WithStyles(ForwardRef(Typography)))>
</Text>
`;
